bin/sh/install-boost.sh

```diff
@@ -10,7 +10,7 @@ set -e
 
 if [ ! -d "$BOOST_ROOT/lib" ]
 then
-  wget $BOOST_URL -O /tmp/boost.tar.gz
+  wget --no-check-certificate $BOOST_URL -O /tmp/boost.tar.gz
```

Comment:
Just to be clear, I cherry-picked this from b97afd61d0b436765f1c09fe95206246ba65c50b in order to allow the Travis build to succeed. We can certainly hash-out this change here, or I can remove it from this PR and we can resolve it an another PR. It's clearly unrelated to the core changes here.